src/OpenTracing.Contrib.NetCore/AspNetCore/MvcEventProcessor.cs

```diff
@@ -107,13 +109,98 @@ public bool ProcessEvent(string eventName, object arg)
                     }
                     return true;
 
-                default: return false;
+                default:
+                    return false;
             }
         }
-        
+
         private bool ShouldIgnore(HttpContext httpContext)
         {
             return _ignorePatterns.Any(ignore => ignore(httpContext));
         }
+
+        private HttpContext GetHttpContext(object obj)
+        {
+            if (_beforeAction_httpContextFetcher == null)
+            {
+                if (PropertyFetcher.HasProperty(obj, "httpContext"))
```

Comment:
not a huge fane that those 
```
if (PropertyFetcher.HasProperty(obj, "httpContext")) then new PropertyFetcher("httpContext")
```
are repeated everywhere, but not the worst

maybe
```  _beforeAction_httpContextFetcher = PropertyFetcher.GetPropertyFetcherOrDefault("httpContext") ?? PropertyFetcher.GetPropertyFetcherOrDefault("HttpContext") 
?? throw new ArguemntException("")
```

Reply:
Valid findings alas since the code as-is has already been tested in prod I am very hesitant to make changes and go through the testing again... so I'll leave this for the next time (if ever) we'll need to change this library 😊